latextools_utils/bibcache.py

```diff
@@ -0,0 +1,89 @@
+import os
+import time
+
+import sublime
+
+if sublime.version() < '3000':
+    _ST3 = False
+    from latextools_utils import bibformat, cache, get_setting
+else:
+    _ST3 = True
+    from . import bibformat, cache, get_setting
+
+
+def write_fmt(bib_name, bib_file, bib_entries):
+    cache_name, formatted_cache_name = _cache_name(bib_name, bib_file)
+
+    current_time = time.time()
+
+    # write the full unformatted bib entries into the cache together
+    # with a time stamp
+    print("Writing bibliography into cache {0}".format(cache_name))
+    cache.write_global(cache_name, (current_time, bib_entries))
+
+    # create and cache the formatted entries
+    formatted_entries = _create_formatted_entries(formatted_cache_name,
+                                                  bib_entries, current_time)
+    return formatted_entries
+
+
+def read_fmt(bib_name, bib_file):
+    cache_name, formatted_cache_name = _cache_name(bib_name, bib_file)
+
+    try:
+        meta_data, formatted_entries = cache.read_global(formatted_cache_name)
+    except:
+        raise cache.CacheMiss()
+
+    # raise a cache miss if the modification took place after the caching
+    modified_time = os.path.getmtime(bib_file)
+    if modified_time > meta_data["cache_time"]:
+        raise cache.CacheMiss()
+
+    # validate the format string are still valid
+    if any(meta_data[s] != get_setting("cite_" + s)
```

Comment:
No. I could image the bibliography could change and sanitize the strings, e.g. the author name from `Bj{\"o}rn` to `Björn`. Or it could somehow increase the precision, e.g. the traditional could find more authors.
However I don't see any real world example and it would not break the compatibility to the old cached entries anyway.